### PR TITLE
EOS-28577: hctl status not able to read 'm0_client_types' key

### PR DESCRIPTION
**Pull request by SwapnilGaonkar7**
**Solution**: Modified key structure to make it dictionary
Aso Modified 'hctl fetch-fids' to include endpoint info.
Also added hax in the list of services returned by 'hctl fetch-fids'.

**Test:**
I ran 'prepare' and 'config' in LR and LC(data and server POD pod). Stages were successfull and following is the 'hctl status' output after changes.

On LC:

[root@cortx-server-headless-svc-ssc-vm-g4-rhev4-0416 tmp1]# hctl status
Byte_count:
    critical_byte_count : 0
    damaged_byte_count : 0
    degraded_byte_count : 0
    healthy_byte_count : 0
Data pool:
    # fid name
    0x6f00000000000001:0x33 'storage-set-1__sns'
Profile:
    # fid name: pool(s)
    0x7000000000000001:0x50 'Profile_the_pool': 'storage-set-1__sns' 'storage-set-1__dix' None
Services:
    cortx-server-headless-svc-ssc-vm-g4-rhev4-0416
    [started]  hax        0x7200000000000001:0x2d  inet:tcp:cortx-server-headless-svc-ssc-vm-g4-rhev4-0416@2001
    [started]  rgw        0x7200000000000001:0x30  inet:tcp:cortx-server-headless-svc-ssc-vm-g4-rhev4-0416@5001
    cortx-data-headless-svc-ssc-vm-g4-rhev4-0416
    [started]  hax        0x7200000000000001:0x7   inet:tcp:cortx-data-headless-svc-ssc-vm-g4-rhev4-0416@2001
    [started]  ioservice  0x7200000000000001:0xa   inet:tcp:cortx-data-headless-svc-ssc-vm-g4-rhev4-0416@3001
    [started]  ioservice  0x7200000000000001:0x19  inet:tcp:cortx-data-headless-svc-ssc-vm-g4-rhev4-0416@3002
    [started]  confd      0x7200000000000001:0x28  inet:tcp:cortx-data-headless-svc-ssc-vm-g4-rhev4-0416@3003

On LR:
[root@ssc-vm-g2-rhev4-2221 cortx-hare]# hctl status
Byte_count:
    critical_byte_count : 0
    damaged_byte_count : 0
    degraded_byte_count : 0
    healthy_byte_count : 0
Data pool:
    # fid name
    0x6f00000000000001:0x3d 'the pool'
Profile:
    # fid name: pool(s)
    0x7000000000000001:0x5b 'default': 'the pool' None None
Services:
    localhost  (RC)
    [started]  hax        0x7200000000000001:0x6   inet:tcp:192.168.59.148@2001
    [started]  confd      0x7200000000000001:0x9   inet:tcp:192.168.59.148@3001
    [started]  ioservice  0x7200000000000001:0xc   inet:tcp:192.168.59.148@3002
    [unknown]  rgw        0x7200000000000001:0x2b  inet:tcp:192.168.59.148@5001
    [unknown]  mgw        0x7200000000000001:0x2e  inet:tcp:192.168.59.148@5002
    [unknown]  mgw        0x7200000000000001:0x31  inet:tcp:192.168.59.148@5003
    [unknown]  mgw        0x7200000000000001:0x34  inet:tcp:192.168.59.148@5004
    [unknown]  s3server   0x7200000000000001:0x37  inet:tcp:192.168.59.148@5005
    [unknown]  s3server   0x7200000000000001:0x3a  inet:tcp:192.168.59.148@5006

### CODE REVIEW

**Review by vaibhavparatwar — APPROVED** [review]

File: utils/hare-status [hunk 1]
```diff
@@ -177,9 +177,20 @@ def processes(cns: Consul, consul_util: ConsulUtil,
     node_health = consul_util.get_node_health_status(node_name)
     process_states = get_node_process_states(cns, node_name)
     process_list = get_node_processes(cns, node_name)
+
```
Comment (suvratjoshi) [inline utils/hare-status:180]: Do we need to have decorator `@repeat_if_fails` for this function processes()?

Comment (SwapnilGaonkar7) [reply utils/hare-status:180]: @suvratjoshi Above function is getting called from functions(**show_text_status** and **get_cluster_status**) and we have '@repeat_if_fails' for both of them

**Review by suvratjoshi — APPROVED** [review]

**Comment by SwapnilGaonkar7** [comment]
@mssawant I have added new commit for improving 'hctl fetch-fids'.
I have tested deployment and following are the results
```
[root@cortx-server-headless-svc-ssc-vm-g4-rhev4-0635 /]# hctl status
Byte_count:
    critical_byte_count : 0
    damaged_byte_count : 0
    degraded_byte_count : 0
    healthy_byte_count : 0
Data pool:
    # fid name
    0x6f00000000000001:0x33 'storage-set-1__sns'
Profile:
    # fid name: pool(s)
    0x7000000000000001:0x50 'Profile_the_pool': 'storage-set-1__sns' 'storage-set-1__dix' None
Services:
    cortx-server-headless-svc-ssc-vm-g4-rhev4-0635
    [started]  hax        0x7200000000000001:0x2d  inet:tcp:cortx-server-headless-svc-ssc-vm-g4-rhev4-0635@2001
    [started]  rgw        0x7200000000000001:0x30  inet:tcp:cortx-server-headless-svc-ssc-vm-g4-rhev4-0635@5001
    cortx-data-headless-svc-ssc-vm-g4-rhev4-0635  (RC)
    [started]  hax        0x7200000000000001:0x7   inet:tcp:cortx-data-headless-svc-ssc-vm-g4-rhev4-0635@2001
    [started]  ioservice  0x7200000000000001:0xa   inet:tcp:cortx-data-headless-svc-ssc-vm-g4-rhev4-0635@3001
    [started]  ioservice  0x7200000000000001:0x19  inet:tcp:cortx-data-headless-svc-ssc-vm-g4-rhev4-0635@3002
    [started]  confd      0x7200000000000001:0x28  inet:tcp:cortx-data-headless-svc-ssc-vm-g4-rhev4-0635@3003
```

```
[root@cortx-server-headless-svc-ssc-vm-g4-rhev4-0635 /]# hctl fetch-fids -c /etc/cortx/hare/config/7c82776696c44674ae6192d3b9702cf0/ --all
[
  {
    "name": "cortx-data-headless-svc-ssc-vm-g4-rhev4-0635",
    "svcs": [
      {
        "name": "hax",
        "fid": "0x7200000000000001:0x7",
        "ep": "inet:tcp:cortx-data-headless-svc-ssc-vm-g4-rhev4-0635@2001"
      },
      {
        "name": "ioservice",
        "fid": "0x7200000000000001:0xa",
        "ep": "inet:tcp:cortx-data-headless-svc-ssc-vm-g4-rhev4-0635@3001"
      },
      {
        "name": "ioservice",
        "fid": "0x7200000000000001:0x19",
        "ep": "inet:tcp:cortx-data-headless-svc-ssc-vm-g4-rhev4-0635@3002"
      },
      {
        "name": "confd",
        "fid": "0x7200000000000001:0x28",
        "ep": "inet:tcp:cortx-data-headless-svc-ssc-vm-g4-rhev4-0635@3003"
      }
    ]
  },
  {
    "name": "cortx-server-headless-svc-ssc-vm-g4-rhev4-0635",
    "svcs": [
      {
        "name": "hax",
        "fid": "0x7200000000000001:0x2d",
        "ep": "inet:tcp:cortx-server-headless-svc-ssc-vm-g4-rhev4-0635@2001"
      },
      {
        "name": "rgw",
        "fid": "0x7200000000000001:0x30",
        "ep": "inet:tcp:cortx-server-headless-svc-ssc-vm-g4-rhev4-0635@5001"
      }
    ]
  }
]
```

**Comment by mssawant** [comment]
@SwapnilGaonkar7, kindly update the commit message as,

```
[EOS-28577: hctl fetch-fids does not include hax information

hctl fetch-fids presently returns only the fids of the configured motr processes
on a given node or cluster. Motr components (server and client) components
fetch additional information; viz:- hax fid and endpoint, from separately
maintained configuration files, typically located at `/etc/sysconfig`. Therefore,
`hctl fetch-fids` command can be further improved to also report `hax` fid and
endpoint along with services fids.
 
Solution: 
- Updated 'hctl fetch-fids' to include hax fid and endpoint information.
- Added hax in the list of services supported by 'hctl fetch-fids' command.
```

**Comment by mssawant** [comment]
Also for the earlier commit,
```
[EOS-28577: hctl status not able to read 'm0_client_types' key

Solution: 
updated key (`please mention the corresponding old kv`) to ('please mention the new kv`)
```

File: utils/update-consul-conf [hunk 2]
```diff
@@ -250,6 +250,7 @@ IOS_IDs=$(get_service_ids_from_kv_file 'grep -iw "services\/ios"')
 
 MOTR_CLIENT_TYPES=$(get_motr_client_types)
 MOTR_CLIENT_TYPES=$(echo $MOTR_CLIENT_TYPES  | cut -d "[" -f2 | cut -d "]" -f1)
+MOTR_CLIENT_TYPES=$(echo $MOTR_CLIENT_TYPES  | cut -d "\\" -f2 | cut -d "\"" -f2)
```
[the remark this answers is not in the document]
Comment (SwapnilGaonkar7) [reply utils/update-consul-conf:253]: Sure, I will try `jq` here

**Comment by vaibhavparatwar** [comment]
@mssawant if changes are done by @SwapnilGaonkar7  can you re-review it ? and merge if all looks good?

**Comment by SwapnilGaonkar7** [comment]
@vaibhavparatwar I am still addressing review comment by Mandar

File: utils/update-consul-conf (hunk 2 above)
[the remark this answers is not in the document]
Comment (SwapnilGaonkar7) [reply utils/update-consul-conf]: Done

**Comment by SwapnilGaonkar7** [comment]
@mssawant I have addressed review comments.
Tested deployment after changes.
Please have a look

**Review by mssawant — APPROVED** [review]

**Pull request merged**